src/greeter/GreeterApp.cpp

```diff
@@ -231,6 +234,29 @@ namespace SDDM {
         view->deleteLater();
     }
 
+    void GreeterApp::pamStoreMessage(const QString message) {
+        m_pamMessages.append(message);
+    }
+
+    void GreeterApp::pamShow() {
+        while (!m_pamMessages.isEmpty()) {
+            QMessageBox *m = new QMessageBox(nullptr);
+            m->setWindowTitle(tr("PAM Message"));
+            m->setText(m_pamMessages.takeFirst());
+            m->exec();
+        }
+    }
+
+    void GreeterApp::pamShowCloseGreeter() {
+        // closing greeter-views to show user's window manager
+        while (!m_views.isEmpty()) {
+            m_views.takeFirst()->close();
+        }
+
+        // display PAM messages
+        pamShow();
```

Comment:
so the timeline is:

 - we close the greeter
 - we spawn the user's session
 - we spawn dialogs as the SDDM user in the user's session